addons/gift_card/models/gift_card.py

```diff
@@ -50,7 +51,7 @@ def balance_converted(self, currency_id=False):
 
     def can_be_used(self):
         # expired state are computed once a day, so can be not synchro
-        return self.state == 'valid' and self.balance > 0 and self.expired_date >= fields.Date.today()
+        return self.state == 'valid' and self.balance > 0 and fields.Date().today() + timedelta(days=1) if not self.expired_date else self.expired_date >= fields.Date.today()
```

Comment:
you may directly check the boolean value
`self.state == 'valid' and self.balance > 0 and (not self.expired_date or self.expired_date >= fields.Date.today())`